Dont ignore adding to index exception or dumpFile error

ravyolii opened the pull request:
Part of LIM-31935

We run with dry run double check, and when there's an error in addToEntry, we just break - causing the counts between dryRun and non dryRun to be incorrect, and thus we anyway terminate the process with an error.

First, need to log the actual error so we'll know there are issues in the index writing code.

Second, once we can verify there are no double checks errors i want to remove this code, it adds complexity, and I don't think it will be needed once all errors are accounted for.

barapiiro commented on git/git.go:
```diff
@@ -385,7 +393,7 @@ func (provider *repositoryProvider) snapshot(repository *git.Repository, commit
 			err = addEntryToIndexFile(indexOutputFile, name, &entry)
 			if err != nil {
-				break
+				return 0, fmt.Errorf("failed to write to index file for '%v': %v", name, err)
```
you missed one break above

ravyolii replied:
OHhh ok, i found the issue here with all the breaks, there was that there's an additional 'err' variable thats hiding the outer scope

ravyolii commented:
I changed the code - there was strange pattern, when it did 'break' and then checked the 'err' variable... but there was an inner err hiding the outer scope variable.
I changed it to exit the loop on err where its relevant. Then I was able to remove some unreachable code.
